test/benchmark.py

```diff
@@ -0,0 +1,327 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A test suite to check Spanner dialect for SQLAlchemy performance
+in comparison with the original Spanner client.
+"""
+import datetime
+import random
+from scipy.stats import sem
+import statistics
+import time
+
+from google.cloud import spanner_dbapi
+from google.cloud.spanner_v1 import Client, KeySet
+from sqlalchemy import (
+    create_engine,
+    insert,
+    select,
+    text,
+    MetaData,
+    Table,
+)
+
+
+def measure_execution_time(function):
+    """Decorator to measure a wrapped method execution time."""
+
+    def wrapper(self, measures):
+        """Execute the wrapped method and measure its execution time.
+
+        Args:
+            measures (dict): Test cases and their execution time.
+        """
+        t_start = time.time()
+        try:
+            function(self)
+            measures[function.__name__] = round(time.time() - t_start, 2)
+        except Exception:
```

Comment:
Catching all exceptions is bad practice as it makes it difficult to know when things go wrong. I would much prefer you catch on specific error such as `Aborted` or `DeadlineExceeded` or whatever. Currently, if there is an issue relating to a table or row already existing, there is no indication of this.

Reply:
Well, as this is a performance benchmark, not a functionality test, I've decided to catch all the errors. However, true, it can cause some troubles. I think the only annoying exception right now is `Aborted`. Will see if others will appear.